cashu/wallet/wallet.py

```diff
@@ -999,6 +1017,23 @@ async def create_p2sh_lock(self):
         await store_p2sh(p2shScript, db=self.db)
         return p2shScript
 
+    async def create_p2pk_lock(self):
+        public_key = PrivateKey(
+            bytes.fromhex(settings.nostr_private_key), raw=True
+        ).pubkey
+        assert public_key
+        return public_key.serialize().hex()
+
+    async def sign_p2pk_with_privatekey(self, proofs: List[Proof]) -> List[str]:
+        private_key = PrivateKey(bytes.fromhex(settings.nostr_private_key), raw=True)
+        assert private_key.pubkey
+        return [
+            sign_p2pk_sign(
+                message=proof.secret.encode("utf-8"), private_key=private_key
+            )
+            for proof in proofs
+        ]
+

```

Comment:
Receiver: user nostr pubkey as receiving key and private key for signing during redemption.